snoop/data/analyzers/entities.py

```diff
@@ -244,11 +308,22 @@ def get_entity_results(blob, language=None, translation_result_pk=None):
 
     if digest_data.get('text'):
         text_sources['text'] = digest_data.get('text', '')[:text_limit]
-
-    if digest_data.get('ocrtext'):
+    tesseract_lang_code = LANGUAGE_CODE_MAP.get(language)
+    if not tesseract_lang_code:
+        log.warning(f'No OCR Language code found for language: {language}')
+        # keep all texts if there is no matching language code
+        tesseract_lang_code = ""
+    if digest_data.get('ocrtext') and tesseract_lang_code in current_collection().ocr_languages:
         for k, v in digest_data.get('ocrtext').items():
+            if not k.endswith(tesseract_lang_code):
```

Comment:
same as other comment: check for whole name `tesseract_{lang}` instead of checking the suffix. Custom names may exist in this module.